py/desispec/scripts/master_nea.py

```diff
@@ -0,0 +1,125 @@
+"""
+This script generates a master NEA (Noise Equivalent Area) file for a given camera.
+"""
+from   specter.psf.gausshermite  import  GaussHermitePSF
+from   desiutil.log import get_logger
+from   desispec.parallel import default_nproc as numprocesses
+from   functools import partial
+
+import numpy as np
+import argparse
+import sys
+import copy
+import astropy.io.fits as fits
+
+# https://github.com/desihub/desispec/issues/1006
+sampling           = 500
+
+wmin, wmax, wdelta = 3600., 9824., 0.8
+fullwave           = np.round(np.arange(wmin, wmax + wdelta, wdelta), 1)
+cslice             = {"b": slice(0, 2751, sampling), "r": slice(2700, 5026, sampling), "z": slice(4900, 7781, sampling)}
+
+def parse(options=None):
+    parser = argparse.ArgumentParser(description="Generate master NEA file for a given camera.")
+    parser.add_argument('-i','--infile', type = str, default = None, required=True,
+                        help = 'path of DESI psf fits file.')
+    parser.add_argument('--outdir', type = str, default = None, required=True,
+			help = 'dir. of output maser nea file for a given camera')
+    args = None
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+def process_one(w, psf, ifiber):    
+    psf_2d = psf.pix(ispec=ifiber, wavelength=w)
+    psf_1d = np.sum(psf_2d, axis=0)
+    
+    # Normalized to one by definition (TBC, again).                                                                                                                                                                                    
+    # dA = 1.0 [pixel units]                                                                                                                                                                                                           
+    norm = np.sum(psf_1d)
+    psf_1d /= norm
+    
+    try:
+        # Automatically raises an assertion.                                                                                                                                                                                              
+        # np.testing.assert_almost_equal(norm, 1.0, decimal=7)
+
+        # http://articles.adsabs.harvard.edu/pdf/1983PASP...95..163K                                                                                                                                                                
+        nea       = 1. / np.sum(psf_1d**2.)  # [pixel units].                                                                                                                                                                           
+        angperpix = psf.angstroms_per_pixel(ifiber, w)
+
+        return  [nea, angperpix]
+
+    except:
+        print('Failed on fiber {} and wavelength {} [{} to {} limit] with norm {}.'.format(ifiber, w, psf._wmin_spec[ifiber], psf._wmax_spec[ifiber], norm))        
+        return [-99., -99.]
```

Comment:
Is this try/except needed, or leftover from testing.  If still needed, let's use the logger instead of a bare print.

Reply:
This is catching the un-normalized psf values as erroneous and patching.  Will change to logger.